### PR TITLE
Only make necessary changes

### PR DESCRIPTION
**Pull request by jacksonj04**
When someone is removed from a group in Zendesk, all tickets assigned to them as part of that group membership are unassigned from the individual and reassigned to the group. Whilst this is intended behaviour when someone leaves their support rota, we should not be unassigning people mid-rota.

This PR introduces checks that the action of either removing or adding a person from the support group is necessary to make the group match the rota.

### CODE REVIEW

File: lib/huntaway.rb [hunk 1]
```diff
@@ -9,22 +9,31 @@ class Huntaway
...
+    assign_missing_support_users_to_group
   end
 
   private
 
-  def assign_current_support_users_to_group
+  def assign_missing_support_users_to_group
     current_support_user_ids.each do |id|
-      client.group_memberships.create!(user_id: id, group_id: FIRST_LINE_DEV_SUPPORT_GROUP_ID)
+      unless @@existing_group_memberships.include?(id)
```
Comment (pezholio) [inline lib/huntaway.rb]: What's the rationale behind making this a class variable?

Comment (jacksonj04) [reply lib/huntaway.rb]: A class variable means we can populate it as part of the process of deleting (which already includes the logic for if a membership is part of the right group or not). I suppose we could easily enough pull it out into a method and isolate that bit of logic.

Comment (pezholio) [reply lib/huntaway.rb]: Yeah, but it's all taking place within one instance of the same class, so only needs to be an instance variable right?

**Review by pezholio — APPROVED** [review]
LGTM 👍

**Pull request merged**